apps/server/src/modules/authorization/domain/rules/board-do.rule.ts

```diff
@@ -16,30 +23,129 @@ export class BoardDoRule implements Rule {
 
 	public hasPermission(user: User, boardDoAuthorizable: BoardDoAuthorizable, context: AuthorizationContext): boolean {
 		const hasPermission = this.authorizationHelper.hasAllPermissions(user, context.requiredPermissions);
-		if (hasPermission === false) {
+		if (!hasPermission) {
 			return false;
 		}
 
-		const userBoardRole = boardDoAuthorizable.users.find(({ userId }) => userId === user.id);
-		if (!userBoardRole) {
+		const userBoardRoles = boardDoAuthorizable.users.find(({ userId }) => userId === user.id);
+		if (!userBoardRoles) {
 			return false;
 		}
 
-		if (boardDoAuthorizable.requiredUserRole && boardDoAuthorizable.requiredUserRole !== userBoardRole.userRoleEnum) {
+		if (this.shouldProcessParentDo(boardDoAuthorizable)) {
```

Comment:
It's hard to read if you cut the function by parent and child.

Make the cut by use case.

Like
handleSubmission(...);
handleDrawingElement(...);
handleElse(...);

This way the reader is aware of the goal of the method and can scope them mentally.

Right now I have to carry a lot of context between the reads of the different methods to understand what is happening.